src/core/config/netdata_publisher.h

```diff
@@ -0,0 +1,110 @@
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes compile-time configurations for Network Data Publisher.
+ *
+ */
+
+#ifndef CONFIG_NETDATA_PUBLISHER_H_
+#define CONFIG_NETDATA_PUBLISHER_H_
+
+#include "config/srp_server.h"
+
+/**
+ * @def OPENTHREAD_CONFIG_NETDATA_PUBLISHER_ENABLE
+ *
+ * Define to 1 to enable Network Data Publisher.
+ *
+ * Network Data Publisher provides mechanisms to limit the number of similar Service and/or Prefix (on-mesh prefix
+ * or external route) entries in the Thread Network Data by monitoring the Network Data and managing if or when to add
+ * or remove entries.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_NETDATA_PUBLISHER_ENABLE
+#define OPENTHREAD_CONFIG_NETDATA_PUBLISHER_ENABLE OPENTHREAD_CONFIG_SRP_SERVER_ENABLE
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_NETDATA_PUBLISHER_MAX_WAIT_TIME_TO_ADD
+ *
+ * Specifies the maximum value (in milliseconds) of the random wait time used before adding an entry in the Thread
+ * Network Data.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_NETDATA_PUBLISHER_MAX_WAIT_TIME_TO_ADD
+#define OPENTHREAD_CONFIG_NETDATA_PUBLISHER_MAX_WAIT_TIME_TO_ADD 5000
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_NETDATA_PUBLISHER_MAX_WAIT_TIME_TO_REMOVE
+ *
+ * Specifies the maximum value (in milliseconds) of the random wait time used before removing an entry from the Thread
+ * Network Data.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_NETDATA_PUBLISHER_MAX_WAIT_TIME_TO_REMOVE
+#define OPENTHREAD_CONFIG_NETDATA_PUBLISHER_MAX_WAIT_TIME_TO_REMOVE 20000
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_NETDATA_PUBLISHER_EXTRA_WAIT_TIME_TO_REMOVE_PREFERRED
+ *
+ * Specifies the extra wait time (in milliseconds) used when removing a preferred entry.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_NETDATA_PUBLISHER_EXTRA_WAIT_TIME_TO_REMOVE_PREFERRED
+#define OPENTHREAD_CONFIG_NETDATA_PUBLISHER_EXTRA_WAIT_TIME_TO_REMOVE_PREFERRED 20000
```

Comment:
Should `OPENTHREAD_CONFIG_NETDATA_PUBLISHER_EXTRA_WAIT_TIME_TO_REMOVE_PREFERRED` always be >= `OPENTHREAD_CONFIG_NETDATA_PUBLISHER_MAX_WAIT_TIME_TO_REMOVE`?

Reply:
Ideally yes (if we want to ensure the preferred entry is removed after giving chance to all non-preferred one) but I don't think we want to check and require this (can recommend it in description of this config).